kore/src/Kore/Equation/Application.hs

```diff
@@ -368,16 +368,22 @@ checkRequires sideCondition predicate requires =
         let requires' = makeAndPredicate predicate requires
             -- The condition to refute:
             condition :: Condition RewritingVariableName
-            condition = from @(Predicate _) (makeNotPredicate requires')
-        return condition
-            -- First try to refute 'condition' without user-defined axioms:
-            >>= withoutAxioms . simplifyCondition
-            -- Next try to refute 'condition' including user-defined axioms:
-            >>= withAxioms . simplifyCondition
-            -- Finally, try to refute the simplified 'condition' using the
-            -- external solver:
-            >>= SMT.filterBranch . withSideCondition
-            >>= return . snd
+            condition = from @(Predicate _) $ makeNotPredicate requires'
+
+        -- First try to refute 'condition' without user-defined axioms:
+        simpleConditionWithoutAxioms <- withoutAxioms $ simplifyCondition condition
+        -- Next try to refute 'condition' including user-defined axioms:
+        simpleConditionWithAxioms <- simplifyCondition simpleConditionWithoutAxioms
```

Comment:
This is the expanded pipeline, it seems that we run `simplifyCondition` without actually adding the axioms back in. This is obfuscated in the older code due to the pipelined indirection. Is this correct?

Reply:
This needs to be reverted back to old pipeline, there is no issue it is correct.

Reply:
Reverted in commit https://github.com/kframework/kore/pull/2864/commits/98980e6c59739ec4da924156b4dc9d97dabfbda7